### ISSUE TITLE
 Web - Chat disappears after being sent

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#21883]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open any DM with another user
2. Send a message
3. Note after a few minutes that the message is no longer there

## Expected Result:
The message is sent.

## Actual Result:
The message has disappeared.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.33-4
**Reproducible in staging?:** n/a
**Reproducible in production?:** n/a
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

his occurred late last night and I didn’t catch it happening to record a screenshot. I haven’t been able to reproduce it this morning, but multiple guides have reported messages disappearing after sending their customer #admins rooms. The [logs](https://www.expensify.com/_devportal/tools/logSearch/#query=request_id:(%227de029d69c1b8861-LHR%22)+AND+timestamp:%5B2023-06-27T18:40:44.100Z+TO+2023-06-27T20:40:44.100Z%5D&index=logs_expensify-023011) from the time I attempted to send the message (19:39 UTC 27/06/23) say:

`[hmmm] [PusherConnectionManager] Channels Error 1006 ~~ error: '[type: 'PusherError' data: '[code: '1006' message: 'The operation couldn't be completed. Socket is not connected']']' userAgent: 'New%20Expensify/1.3.31.3 CFNetwork/1408.0.4 Darwin/22.5.0'`
 `[hmmm] [PusherConnectionManager] WebSocketError ~~ error: '[type: 'WebSocketError' error: '[isTrusted: '' message: 'The operation couldn't be completed. Socket is not connected']']' userAgent: 'New%20Expensify/1.3.31.3 CFNetwork/1408.0.4 Darwin/22.5.0'`

**Expensify/Expensify Issue URL:**
**Issue reported by:** Milja Kljajic
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1687969003963889

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Issue Owner</summary>Current Issue Owner: @mountiny</details>

**Comment by kadiealexander** [comment]
Have asked about steps to take to reproduce this issue [here](https://expensify.slack.com/archives/C01SKUP7QR0/p1688363747696039?thread_ts=1686024498.712479&cid=C01SKUP7QR0).

**Comment by kadiealexander** [comment]
Assigning @iwiznia to help with nailing down the reproduction of this via logs, as per [this thread.](https://expensify.slack.com/archives/C049HHMV9SM/p1687970892569839?thread_ts=1687969003.963889&cid=C049HHMV9SM) Please let me know if I should approach this a different way Ionatan, I'm a bit stumped on how to reliably reproduce this!

**Comment by iwiznia** [comment]
ok device logs are here https://www.expensify.com/_devportal/tools/logSearch/#query=blob%3A%22AuthenticatePusher%22%20AND%20email%3A%22k.alexander%40expensify.com%22%20AND%20timestamp%3A%5B2023-07-03T00%3A00%20TO%202023-07-06T23%3A59%5D&index=_all
You can see it calling AuthenticatePusher over and over
Then if you look at the logs from the server, you see that it fails with 407 authToken expired: https://www.expensify.com/_devportal/tools/logSearch/#query=request_id%3A(%227e0c993709e5ee9e-SJC%22)%20AND%20timestamp%3A%5B2023-07-03T04%3A06%3A19.503Z%20TO%202023-07-03T06%3A06%3A19.503Z%5D&index=_all
I am assuming pusher calls go through a different flow that do not handle authToken expiring properly... need to investigate a bit more.

**Comment by iwiznia** [comment]
Fuck I totally forgot about this, will try to work on it or ask someone to take it

**Comment by kadiealexander** [comment]
@iwiznia any thoughts on this one? Should we find someone else who might have some bandwidth?

**Comment by iwiznia** [comment]
I actually can't reproduce the error you were getting. I tried forcing authenticate pusher to return 407 but that is triggering the authentication correctly and not getting stuck in a loop.
Long story short, I don't know what the problem was....

**Comment by kadiealexander** [comment]
It's not my error I just got the Bug0 assignment, I'm not able to reproduce this either. 

@miljakljajic is this still happening for you?

**Comment by miljakljajic** [comment]
I think it's related to having a ton (thousands) of newDot policies which is why you're struggling to reproduce it. I think we discussed having Applause test these types of wave 1 support bugs by setting up accounts with a similar number of policies?

**Comment by iwiznia** [comment]
Can you reproduce the issue now @miljakljajic or was it a one off?

**Comment by miljakljajic** [comment]
Not for a couple of weeks at least but I've asked in #guides as I wasn't the only person experiencing this issue.

**Comment by miljakljajic** [comment]
Stilll reproducible for at least three of our Guides:

**Describe the exact steps you took to arrive at the issue:** I sent a message to my user yesterday in which he stated he did not receive. I can see the message in public mode however, I do not see the message in the incognito mode. Now it's stating I do not have access to the chat via incognito.
**Your email:** [elle.jackson@team.expensify.com](mailto:elle.jackson@team.expensify.com)
**Platform** (e.g. desktop app, chrome, iOS, android): Desktop
**App version:** v1.3.43-7
**Report ID:** 3975761511991100
**Screenshots of what you’re seeing in the app & Console -**
![image (42)](https://github.com/Expensify/App/assets/28715117/7ae1875b-32d2-4c9e-a0c5-63fef661dade)
![image (41)](https://github.com/Expensify/App/assets/28715117/84366fbb-4643-44d7-a548-70fef8da5146)
![image (40)](https://github.com/Expensify/App/assets/28715117/533162a9-fc38-4065-ab92-abf2e2885e75)
![image (39)](https://github.com/Expensify/App/assets/28715117/2fd52a63-2b49-4be5-a215-4a7802dae734)

**Comment by iwiznia** [comment]
Damn, I am very lost on the issue. More investigation is needed, but I am going on sabbatical, so please reassign another engineer

**Comment by kadiealexander** [comment]
Not overdue! @mountiny we have a weird issue affecting our Guides teams, and it's almost impossible to reproduce reliably. Could you please look into this and see if there's anything we can do (such as adding logs) to narrow down this bug?

**Comment by mountiny** [comment]
From what I could skim over I think we should wait for the reliable updates to be in place and also the onyx improvememnts Margelo will plan to add, seems like really niche bug, combination of potentially missed pusher updates and some flakiness in Onyx due to high amount of stuff to store which margelo is looking into

**Comment by miljakljajic** [comment]
Niche as in rare? Because it has affected at least 4 people that we know of, and probably more people who never noticed that the message wasn't sent.

**Comment by mountiny** [comment]
@miljakljajic Sorry, yes, niche in a sense that its rare and its not clear how to reproduce reliably. We can wait on the reliable updates and onyx improvements to land in app and see if this will still be experienced by some. 

The wave1 improvements should also make this better for guides so I would prefer to wait before spending too much time on this if it will self resolve.

**Comment by kadiealexander** [comment]
Still on hold.

**Comment by mountiny** [comment]
We are close but still on hold

**Comment by miljakljajic** [comment]
Happened again this week - I responded to this customer's message on the 21st on web, but the message didn't send. 

![WhatsApp Image 2023-08-24 at 09 47 06](https://github.com/Expensify/App/assets/28715117/2e1a4e5a-8886-4069-95e8-5ebe3b130009)

**Comment by mountiny** [comment]
@miljakljajic has it been only that one particular message and you were able to successfully send messages to other users around the same time?

**Comment by miljakljajic** [comment]
I actually can't remember which other messages I sent around the same time to confirm. It's possible this was the only message, because I use email as much as possible now.

**Comment by miljakljajic** [comment]
Ah, I see. Strange because I haven't changed timezone, or name, this week...

**Comment by miljakljajic** [comment]
I just noticed another instance of this - a message that I sent to dk@oci-group.co.uk on Aug 18th also didn't send: 

<img width="1440" alt="image" src="https://github.com/Expensify/App/assets/28715117/1c55560d-73c4-4e6c-a056-e3cdf1f769da">

**Comment by miljakljajic** [comment]
I sent my 10am around 7:30 this morning and again the message hasn't gone through:

<img width="1440" alt="image" src="https://github.com/Expensify/App/assets/28715117/dd488a17-c384-447c-8e68-244971416063">

**Comment by mountiny** [comment]
Reliable updates are almost out, I think this might also be related to just large accounts and bad performance.

**Comment by mountiny** [comment]
This still doesn't not have a clear reproduction steps

**Comment by mountiny** [comment]
I have not experienced this and I dont think we had any more widespread or helpful information for how to reproduce.

I thin I would prefer to close this issue in sake of clean ing up plate and focusing on current strategic goals.

Feel free to reopen if this is reproducible
